aws/aws-controllers-k8s
# Crossplane AWS Provider API type generation

Description of changes: As part of https://github.com/crossplane/crossplane/issues/1845 , we will collaborate on getting Crossplane AWS Provider generated using ACK code generation pipeline. This is the very first PR that enables ACK to generate AWS Provider types 🎉 

It is a bit polished version of @jaypipes 's code to generate a Crossplane API type, which is in his [`crossplane`](https://github.com/jaypipes/aws-controllers-k8s/tree/crossplane) branch. We'll follow up with templates to generate Crossplane controllers for available AWS services and work towards adding new services.

cc @jaypipes 

To test, run the following:
```console
$ go run -tags codegen cmd/ack-generate/main.go crossplane apis ecr --provider-dir <directory for provider>
$ cd <directory for provider>
$ go generate ./...
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
